.github/workflows/rust.yml

```diff
@@ -60,14 +60,22 @@ jobs:
   test:
     needs: diff
     if: needs.diff.outputs.isRust == 'true'
-    runs-on: [self-hosted, self-hosted-ubuntu]
+    runs-on: ${{ matrix.os }}
     strategy:
+      matrix:
+        os: [[self-hosted, self-hosted-ubuntu], windows-latest]
       fail-fast: false
     steps:
       - uses: actions/checkout@v3
       - uses: actions-rs/toolchain@v1
       - uses: taiki-e/install-action@nextest
+      - name: cargo test windows
+        continue-on-error: true
+        if: runner.os == 'Windows'
+        run: |
+          cargo nextest run --all-features --profile ci
       - name: cargo test
+        if: runner.os == 'Linux'
```

Comment:
This can be dropped, we can just not block on the windows job via our branch protections